### ISSUE TITLE
Topological / Graph Deep Learning

### ISSUE THREAD

**Issue by KevinBian107** [KevinBian107/TopoVNL#2]
# Considerata

The graph property of the skeleton themselves must be useful, we might be able to use the skeleton for better in discriminating the behaviors?

## Contrastive
- Some type o fGNN with contrastive loss upon W2/kMenas clusters (positive pair = pairs in the same cluster and negative pairs = all other pairs). We can make embeddings of positive pairs closer and negative pairs further. This is kind of building a W2 loss metric (self-supervised where W2/kMeans is pseudo labeling).

- Given a new clip, we can get the cycle and the embedding with DM-g and then just use the model to create a partitioned space where we can place different clips in a reasonable steriotypical behavior position.

## Cross Entropy
- Maybe just using the W2 clsuter as a label and do CrossEntropy loss?
